neo4j/hierarchy_reader.go

```diff
@@ -53,6 +52,42 @@ func (n *Neo4j) GetHierarchyElement(ctx context.Context, instanceID, dimension,
 	return
 }
 
+// HierarchyExists returns true if the hierarchy exists
+func (n *Neo4j) HierarchyExists(ctx context.Context, instanceID, dimension string) (hierarchyExists bool, err error) {
+	neoStmt := fmt.Sprintf(query.HierarchyExists, instanceID, dimension)
+	logData := log.Data{
+		"fn":             "HierarchyExists",
+		"cypher":         neoStmt,
+		"instance_id":    instanceID,
+		"dimension_name": dimension,
+	}
+
+	var vertices []*models.HierarchyElement
+
+	if vertices, err = n.queryElements(ctx, instanceID, dimension, neoStmt, neoArgMap{}); err != nil {
+		if err == driver.ErrNotFound {
+			hierarchyExists = false
+			return hierarchyExists, nil
+		}
+
+		log.Event(ctx, "queryElements failed when attempting to get a hierarchy node", log.ERROR, logData, log.Error(err))
+		return
+	}
+
+	if len(vertices) == 1 {
+		hierarchyExists = true
+		return hierarchyExists, nil
+	}
+
+	if len(vertices) > 1 {
+		err = driver.ErrMultipleFound
+		log.Event(ctx, "expected a single hierarchy node but multiple were returned", log.ERROR, logData, log.Error(err))
+		return hierarchyExists, err
```

Comment:
Wouldn't hierarchyExists be false here but shouldn't it be set to true?

I understand outside this method you would act on the error and not the boolean field but it could be misleading.

Reply:
Good point - it would technically exist. I had only considered the error return value in this case. I will update to return true